src/SegmentedArray.chpl

```diff
@@ -124,17 +124,20 @@ module SegmentedArray {
       var t1 = getCurrentTime();
       ref oa = offsets.a;
       const low = offsets.aD.low, high = offsets.aD.high;
-      // Lengths of segments including null bytes
-      var gatheredLengths: [D] int;
-      forall (gl, idx) in zip(gatheredLengths, iv) with (var agg = new SrcAggregator(int)) {
-        var l: int;
+      // Gather the right and left boundaries of the indexed strings
+      // NOTE: cannot compute lengths inside forall because agg.copy will
+      // experience race condition with loop-private variable
+      var right: [D] int, left: [D] int;
+      forall (r, l, idx) in zip(right, left, iv) with (var agg = new SrcAggregator(int)) {
         if (idx == high) {
-          l = values.size - oa[high];
+          agg.copy(r, values.size);
```

Comment:
I don't think you want to aggregate here. This will end up as something like `const tmp = values.size; agg.copy(r, tmp);` so everything is local. 

What was the original code you wanted to write here?

Reply:
I think this is just for the last value.